libs/types-common/src/Util/Options.hs

```diff
@@ -1,33 +1,57 @@
+{-# LANGUAGE DeriveGeneric   #-}
+{-# LANGUAGE TemplateHaskell #-}
+
 module Util.Options where
 
+import Control.Lens
 import Data.Aeson (FromJSON)
+import Data.Aeson.TH
 import Data.Maybe (fromMaybe)
 import Data.Monoid
-import Data.Yaml (ParseException, decodeFileEither)
+import Data.Text (Text)
+import Data.Word (Word16)
+import Data.Yaml hiding (Parser)
+import GHC.Generics
 import Options.Applicative
 import System.Directory
 import System.Environment (getArgs)
 import System.IO (hPutStrLn, stderr)
+import Util.Options.Common
+
+data Endpoint = Endpoint
+    { _epHost :: !Text
+    , _epPort :: !Word16
+    } deriving (Show, Generic)
+
+deriveFromJSON (toFieldName 3) ''Endpoint
+makeLenses ''Endpoint
+
+data CassandraOpts = CassandraOpts
+    { _casEndpoint :: !Endpoint
+    , _casKeyspace :: !Text
+    } deriving (Show, Generic)
+
+deriveFromJSON (toFieldName 4) ''CassandraOpts
```

Comment:
Perhaps we could get to a more consistent naming scheme for these records/lenses? Dropping [1..4] characters for the json derivation depending on the record is a little brittle, perhaps.